test/framework/src/main/java/org/elasticsearch/transport/nio/channel/NioSocketChannel.java

```diff
@@ -179,11 +173,8 @@ private boolean internalFinish() throws IOException {
 
     private void clearQueuedWrites() {
         // Even if the channel has already been closed we will clear any pending write operations just in case
-        if (state.get() > UNREGISTERED) {
-            SocketSelector selector = getSelector();
-            if (selector != null && selector.isOnCurrentThread() && writeContext.hasQueuedWriteOps()) {
-                writeContext.clearQueuedWriteOps(new ClosedChannelException());
-            }
+        if (socketSelector.isOnCurrentThread() && writeContext.hasQueuedWriteOps()) {
```

Comment:
I see this in several places, how do we get into this? I mean can we maybe turn it into an assertion and make sure only the current thread is calling this method?